ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java

```diff
@@ -196,6 +196,77 @@ public void handleRequests(NetworkRequest networkRequest) throws InterruptedExce
     }
   }
 
+  @Override
+  public void dropRequest(NetworkRequest request) throws InterruptedException {
+    try {
+      InputStream is = request.getInputStream();
+      DataInputStream dis = is instanceof DataInputStream ? (DataInputStream) is : new DataInputStream(is);
+      RequestOrResponseType type = RequestOrResponseType.values()[dis.readShort()];
+      Response response;
+      long requestQueueTime = SystemTime.getInstance().milliseconds() - request.getStartTimeInMs();
+      RequestMetricsUpdater metricsUpdater = new RequestMetricsUpdater(requestQueueTime, 0, 0, 0, true);
+      switch (type) {
```

Comment:
Thanks for creating a function to drop all request types. However, I was wondering if we could return a generic throttling error rather than a request-specific one. The reason is that we can use other frameworks for load shedding, such as Hodor, in the future. Such frameworks will be application-unaware and will likely drop the requests before they make it even this far.